### PR TITLE
DSSMOD-1181

### PR DESCRIPTION
**Pull request by TylerFSmith2**

### CODE REVIEW

File: cypress/e2e/User Account Management/SystemAdminUser.js
Comment (wyntonjones) [inline cypress/e2e/User Account Management/SystemAdminUser.js:1]: remove spaces from folder name.

**Review by wyntonjones — CHANGES_REQUESTED** [review]
The jira ticket I think was copied from the SA automation test, so it has information related to federal users, but the ticket also includes criteria for verifying the user details page (after you select the hyperlink). Should we include those scenarios here or sperate test? I didn't review the SA UAM test, but make the corrections mentioned here in that test if they apply.

File: cypress/e2e/User Account Management/SAOUser.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        userAccountManagement.typeNameSearchInput('elevation');
+        commonPage.clickOnRefreshResultsBtn();
+        userAccountManagement.elements.elevationIcon().should('exist');
+    });
+
+    it("Verify the Refresh Results button is greyed out by default", function () {
+        cy.visit('/User/Account.html');
+        commonPage.elements.refreshResultsBtn().should('be.disabled');
+        userAccountManagement.clickOnSecondaryRoleDropdown();
+        userAccountManagement.elements.secondaryRoleChildren().get('[data-testid="checkbox"]').eq(1).click();
```
[the remark this answers is not in the document]
Comment (TylerFSmith2) [reply cypress/e2e/User Account Management/SAOUser.js:99]: What part of this section do you want defined in the object file?

File: cypress/e2e/User Account Management/SAOUser.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        userAccountManagement.elements.firstRowFourthCol().should('have.text', 'Manager');
+        commonPage.clickOnClearFiltersBtn();
+        userAccountManagement.clickOnSecondaryRoleDropdown();
+        userAccountManagement.elements.secondaryRoleChildren().get('[data-testid="checkbox"]').eq(2).click();
+        commonPage.clickOnRefreshResultsBtn();
+        userAccountManagement.elements.firstRowFourthCol().should('have.text', 'State Authorized Official');
+        commonPage.clickOnClearFiltersBtn();
+        userAccountManagement.clickOnOtherFiltersDropdown();
+        userAccountManagement.elements.otherFiltersChildren().get('[data-testid="checkbox"]').contains('Show Deleted Users').click();
+        userAccountManagement.typeNameSearchInput('Deleted');
```
[the remark this answers is not in the document]
Comment (TylerFSmith2) [reply cypress/e2e/User Account Management/SAOUser.js:80]: In another comment you mention wanting to break up the filters. Let's connect and discuss how these kinds of tests should be approached.

File: cypress/e2e/User Account Management/SAOUser.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        });
+        commonPage.clickOnClearFiltersBtn();
+        userAccountManagement.typeNameSearchInput('Ty');
+        userAccountManagement.elements.firstTableLink().then((text) => {
+            userAccountManagement.typeNameSearchInput('{enter}');
+            userAccountManagement.elements.firstTableLink().contains(text.text()).should('not.exist');
+        });
+        commonPage.clickOnClearFiltersBtn();
+        userAccountManagement.clickOnSecondaryRoleDropdown();
+        userAccountManagement.elements.secondaryRoleChildren().get('[data-testid="checkbox"]').eq(0).click();
```
[the remark this answers is not in the document]
Comment (TylerFSmith2) [reply cypress/e2e/User Account Management/SAOUser.js:64]: Is this talking about the clickOnSecondaryRoleDropdown or secondaryRoleChildren?

[the remark this answers is not in the document]
Comment (TylerFSmith2) [reply cypress/e2e/User Account Management/SAOUser.js:64]: userAccountManagement.elements.secondaryRoleChildren().get('[data-testid="checkbox"]').eq(0)

Change this to a .click function

File: cypress/e2e/User Account Management/SAOUser.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        commonPage.elements.clearFiltersBtn().should('contain', 'Clear Filters');
+        commonPage.elements.refreshResultsBtn().should('have.text', 'Refresh Results');
+        userAccountManagement.elements.tableFirstHeader().should('have.text', 'Username');
+        userAccountManagement.elements.tableSecondHeader().should('have.text', 'First Name');
+        userAccountManagement.elements.tableThirdHeader().should('have.text', 'Last Name');
+        userAccountManagement.elements.tableFifthHeader().should('have.text', 'Secondary Role(s)');
+        userAccountManagement.elements.tableSixthHeader().should('have.text', 'Email');
+        userAccountManagement.elements.tableSeventhHeader().should('have.text', 'Phone');
+    });
+    it("Verify Regional and CB Office Staff users will not be able to access the User Account Management page.", function () {
```
[the remark this answers is not in the document]
Comment (TylerFSmith2) [reply cypress/e2e/User Account Management/SAOUser.js:32]: For above comment - Check to make sure hyperlink is tested. Adjust ticket for both that and removing any SA content.

File: cypress/e2e/UserAccountManagement/SAOUser.js [hunk 5]
```diff
@@ -96,66 +112,95 @@ describe("SAO UAM Page Validations", function () {
         cy.visit('/User/Account.html');
         commonPage.elements.refreshResultsBtn().should('be.disabled');
         userAccountManagement.clickOnSecondaryRoleDropdown();
-        userAccountManagement.elements.secondaryRoleChildren().get('[data-testid="checkbox"]').eq(1).click();
+        userAccountManagement.clickSecondaryRoleCheckbox(1);
         commonPage.elements.refreshResultsBtn().should('not.be.disabled');
     });
-    it("Verify the Manage User Account Requests button works", function () {
+    it.only("Verify the Manage User Account Requests button works", function () {
```
Comment (wyntonjones) [inline cypress/e2e/UserAccountManagement/SAOUser.js:118]: line 118 - remove "only"

**Review by wyntonjones — CHANGES_REQUESTED** [review]

**Review by wyntonjones — APPROVED** [review]

**Pull request merged**